controllers/TokenController.php

```diff
@@ -30,54 +30,58 @@ public function beforeAction($action)
 
     public function actionUpdate()
     {
-        $this->requireLogin();
+        return $this->update(false);
+    }
 
-        $driver = (new DriverService($this->module->getConfigureForm()))->getWebDriver();
-        if (!$driver) {
-            Yii::$app->response->statusCode = 400;
-            return $this->asJson(['success' => false, 'message' => 'No push driver available!']);
-        }
+    public function actionUpdateMobileApp()
+    {
+        return $this->update(true);
+    }
 
-        return $this->asJson([
-            'success' => (
-                (new TokenService())->storeTokenForUser(
-                    Yii::$app->user->getIdentity(),
-                    $driver,
-                    Yii::$app->request->post('token'),
-                )
-            ),
-        ]);
+    public function actionDelete()
```

Comment:
Just to be sure, this action is accessible without login, right? :-)

Reply:
Yes, because I register the FCM assets for guests after logout here: https://github.com/humhub/fcm-push/pull/63/files#diff-666c020a6d1e6a59ba261ba9316aaa19e27dbd851bcc4fdcaecaa278ddc96c10R107